### PR TITLE
Update pfazure/pfs version

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description
- update pfazure version
image

- pfs show promptflow version if installed from root, else devkit version

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/promptflow/azure/_cli/entry.py [hunk 1]
```diff
@@ -140,7 +140,11 @@ def main():
         version_devkit = get_promptflow_devkit_version()
         if version_devkit:
             version_dict["promptflow-devkit"] = version_devkit
-        return json.dumps(version_dict, ensure_ascii=False, indent=2, sort_keys=True, separators=(",", ": ")) + "\n"
+        version_dict_string = (
+            json.dumps(version_dict, ensure_ascii=False, indent=2, sort_keys=True, separators=(",", ": ")) + "\n"
+        )
+        print(version_dict_string)
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/_cli/entry.py:146]: why previously no need print here?

File: src/promptflow-devkit/promptflow/_cli/_pf/_upgrade.py [hunk 2]
```diff
@@ -109,6 +110,7 @@ def upgrade_version(args):
     importlib.reload(json)
 
     version_result = subprocess.check_output(["pf", "version"], shell=platform.system() == "Windows")
+    version_result = version_result.decode().replace("\x1b[0m", "").strip()
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_pf/_upgrade.py:113]: add comment on what's this for?
\x1b[0m

File: src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py [hunk 3]
```diff
@@ -155,6 +155,15 @@ def make_response_no_content():
     return make_response("", 204)
 
 
+def get_pfs_version():
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:158]: add docstring on this

File: src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py [hunk 4]
```diff
@@ -155,6 +155,15 @@ def make_response_no_content():
...
+    version_promptflow = get_promptflow_sdk_version()
+    if version_promptflow:
+        return version_promptflow
+    else:
+        version_devkit = get_promptflow_devkit_version()
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:163]: why shall we do if else here?  I think devkit version should be good enough?

File: src/promptflow-azure/promptflow/azure/_cli/entry.py [hunk 5]
```diff
@@ -140,7 +140,11 @@ def main():
         version_devkit = get_promptflow_devkit_version()
         if version_devkit:
             version_dict["promptflow-devkit"] = version_devkit
-        return json.dumps(version_dict, ensure_ascii=False, indent=2, sort_keys=True, separators=(",", ": ")) + "\n"
+        version_dict_string = (
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/_cli/entry.py:143]: if get_promptflow_sdk_version is None, not include it in the dict

Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/_cli/entry.py:143]: may extract a util function to get the version dict

File: src/promptflow-azure/promptflow/azure/_cli/entry.py (hunk 1 above)
Comment (YingChen1996) [reply src/promptflow-azure/promptflow/azure/_cli/entry.py:146]: not sure, maybe forget change code here to align with pf.
I just update code here to aligh with pf

File: src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py (hunk 4 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:163]: according to the discussion before, we will show promptflow version if customer install from root which is also our recommendation installation method.

File: src/promptflow-azure/promptflow/azure/_cli/entry.py (hunk 5 above)
Comment (YingChen1996) [reply src/promptflow-azure/promptflow/azure/_cli/entry.py]: done

File: src/promptflow-azure/promptflow/azure/_cli/entry.py [hunk 6]
```diff
@@ -123,24 +115,8 @@ def main():
...
-        version_core = get_promptflow_core_version()
-        if version_core:
-            version_dict["promptflow-core"] = version_core
-        # check devkit version
-        version_devkit = get_promptflow_devkit_version()
-        if version_devkit:
-            version_dict["promptflow-devkit"] = version_devkit
-        return json.dumps(version_dict, ensure_ascii=False, indent=2, sort_keys=True, separators=(",", ": ")) + "\n"
+        print_promptflow_version_dict_string(with_azure=True, ignore_none=True)
+        return
```
Comment (0mza987) [inline src/promptflow-azure/promptflow/azure/_cli/entry.py:119]: Why do we have two places to print pf version? Here and line 42. What's the diff between this `print_promptflow_version_dict_string` and `print_pf_version`?

Comment (YingChen1996) [reply src/promptflow-azure/promptflow/azure/_cli/entry.py:119]: I think no difference, only is the print format.

Comment (YingChen1996) [reply src/promptflow-azure/promptflow/azure/_cli/entry.py:119]: extract similar logic to one function

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow-devkit/promptflow/_cli/_pf/_upgrade.py (hunk 2 above)
Comment (crazygao) [reply src/promptflow-devkit/promptflow/_cli/_pf/_upgrade.py:114]: You cannot control this in multiple platforms, remove the color is wiser.

Or else, you can find the run_coverage_test.py and see how to print colored outputs.

File: src/promptflow-devkit/promptflow/_cli/_pf/_service.py [hunk 7]
```diff
@@ -252,20 +252,20 @@ def validate_port(port, force_start):
             subprocess.Popen(cmd, stdout=subprocess.DEVNULL, start_new_session=True)
         is_healthy = check_pfs_service_status(port)
         if is_healthy:
-            message = f"Start Prompt Flow Service on port {port}, version: {get_promptflow_sdk_version()}."
+            message = f"Start Prompt Flow Service on port {port}, version: {get_pfs_version()}."
```
Comment (crazygao) [inline src/promptflow-devkit/promptflow/_cli/_pf/_service.py:255]: Prompt Flow Service -> Promptflow service

Comment (crazygao) [reply src/promptflow-devkit/promptflow/_cli/_pf/_service.py:255]: Also before

File: src/promptflow-azure/promptflow/azure/_cli/entry.py [hunk 8]
```diff
@@ -17,14 +16,7 @@
...
 
-from promptflow._sdk._utils import (  # noqa: E402
-    get_promptflow_azure_version,
-    get_promptflow_core_version,
-    get_promptflow_devkit_version,
-    get_promptflow_sdk_version,
-    get_promptflow_tracing_version,
-    print_pf_version,
-)
+from promptflow._sdk._utils import print_pf_version, print_promptflow_version_dict_string  # noqa: E402
```
Comment (crazygao) [inline src/promptflow-azure/promptflow/azure/_cli/entry.py:19]: no need for noqa?

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_cli/_pf/_upgrade.py (hunk 2 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_cli/_pf/_upgrade.py:114]: "\x1b[0m" is not added by sdk code, but added by termianl

**Review by 0mza987 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**